### ISSUE TITLE
[V2] Navigation.dismissOverlay avoids autoFocus works properly on pushed screen

### ISSUE THREAD

**Issue by manattersa** [wix/react-native-navigation#3891]
### Issue Description

I'm trying to show a message with an overlay before push next screen. When next screen is shown, the keyboard doesn't appear even with a TextInput with autofocus={true}

### Steps to Reproduce / Code Snippets / Screenshots

Two screens, screen A with one TextInput and screen B with another TextInput with autofocus={true},  in A i put some values in the TextInput and then dismiss the keyboard before the Overlay shows up. Next the overlay is dismissed and screen B is pushed, and the Keyboard doesn't shows up.

Apparently, there's an issue in method `- (void)assignKeyWindow` of `RNNOverlayManager` class. Sometimes, the list of windows in the application has 3 instances:

`UIWindow` (the first application window)
`RNNOverlayWindow` (the overlay to be dismissed)
`UITextEffectsWindow` (used by the keyboard)

In `- (void)assignKeyWindow` this list is reversed and then marks the first UIWindow without rootViewController as key. This marks the `UITextEffectsWindow` as the key window, and now keyboard does't show up.

I tried with the following patch:
```
- (void)assignKeyWindow {
	NSArray* windows = [[[UIApplication sharedApplication].windows reverseObjectEnumerator] allObjects];
	for (UIWindow* window in windows) {
		// check if the instance is a UIWindow and not a subclass
		if (window.rootViewController && [window isMemberOfClass:[UIWindow class]]) {
			[window makeKeyAndVisible];
			return;
		}
	}
}
```
And now it works. Maybe the best could be to set a reference to the key window when the overlay is shown, and then make this reference the key window when the overlay is dismissed.

---
### Environment
* React Native Navigation version: 2.0.2511
* React Native version: 0.56.0
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator and Device (iPhone 6), iOS 11.4 and 11.3

**Comment by VicFrolov** [comment]
Any updates on this?

**Comment by Tracklous** [comment]
Still facing this issue in an overlay. Autofocus prop of "TEXTINPUT" is not working in an overlay.

**Comment by ATShiTou** [comment]
Why closed?

**Comment by abury** [comment]
Not sure why this is closed, it's still an issue on 2.24.0.
